### PR TITLE
 planning: parallelize heuristics in reeds shepp path 

### PR DESCRIPTION
**Pull request by deidaraho**

### CODE REVIEW

**Comment by Capri2014** [comment]
@deidaraho  Please do not commit the configs change in.

File: modules/planning/open_space/trajectory_smoother/distance_approach_ipopt_interface.h [hunk 1]
```diff
@@ -30,6 +30,12 @@
 #include "adolc/adolc_sparse.h"
 #include "adolc/adouble.h"
 
+// TO-DO[runxin]: still have issue with adolc parallel
```
Comment (Capri2014) [inline modules/planning/open_space/trajectory_smoother/distance_approach_ipopt_interface.h]: @deidaraho Is this TODO still valid?

Comment (deidaraho) [reply modules/planning/open_space/trajectory_smoother/distance_approach_ipopt_interface.h]: Yes, still there. I need more checking time for this.

File: modules/planning/open_space/trajectory_smoother/distance_approach_ipopt_interface.cc [hunk 2]
```diff
@@ -1399,6 +1400,7 @@ bool DistanceApproachIPOPTInterface::eval_obj(int n, const T* x, T* obj_value) {
 
   *obj_value = 0.0;
   // 1. objective to minimize state diff to warm up
+  // const int max_num_threads = std::max(1,  omp_get_max_threads() - 1);
```
[the remark this answers is not in the document]
Comment (deidaraho) [reply modules/planning/open_space/trajectory_smoother/distance_approach_ipopt_interface.cc]: done

**Pull request merged**